lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs

```diff
@@ -77,23 +86,12 @@ newTransactionLayer
     => Hash "Genesis"
     -> TransactionLayer t k
 newTransactionLayer (Hash block0H) = TransactionLayer
-    { mkStdTx = \keyFrom rnps outs -> do
-        -- NOTE
-        -- For signing, we need to embed a hash of the transaction data
-        -- without the witnesses (since we don't yet have them!). In this sense,
-        -- this is a transaction id as Byron nodes or the http-bridge
-        -- defines them.
-        let inps = fmap (second coin) rnps
-        wits <- forM rnps $ \(_, TxOut addr _) -> do
-            xprv <- maybeToRight (ErrKeyNotFoundForAddress addr) (keyFrom addr)
-            let payload = block0H <> getHash (signData inps outs)
-            pure $ mkTxWitness (fst xprv) (sign payload xprv)
-        let tx = Tx
-                { txId = fragmentId inps outs wits
-                , resolvedInputs = inps
-                , outputs = outs
-                }
-        return (tx, wits)
+    { mkStdTx = mkTokenTransfer MsgTypeTransaction mempty
+
+    , mkCertificateTx = \keyFrom (PoolId poolId, accountXPrv) rnps outs ->
+            let accountPublicKey = CC.unXPub . getRawKey . publicKey $ accountXPrv
```

Comment:
What about requiring the `XPub` version as an argument instead? Seems like demanding a `XPrv` is a bit too strong.

Reply:
We presumably need the `XPrv` for the final "auth" signature.